system.go

```diff
@@ -293,7 +297,7 @@ type System interface {
 	// module.
 	//
 	// Note: This is similar to accept in POSIX.
-	SockAccept(ctx context.Context, fd FD, flags FDFlags) (FD, SocketAddress, Errno)
+	SockAccept(ctx context.Context, fd FD, flags FDFlags) (newfd FD, peer, addr SocketAddress, err Errno)
```

Comment:
Any reason to return the local address here rather than on `SockBind`? Every `SockAccept` call should return the same local address, and it should be known at the time you bind the address (or perhaps listen)? If we can track that information earlier, then we can avoid the extra hit each time an incoming connection is accepted.

I originally thought that you might be able to derive this information without any additional return values, by looking at the input to `SockBind`. However, it's possible for the user to pass in `port=0` and have the OS choose a free port. In this case, it's still useful to have `SockBind` return the actual local address.

Reply:
I originally looked at doing this based on SockBind but it fails for preopen sockets that the application didn't initiate. bind(2) would happen before the module is instantiated and the only way to know the local address is to rely on the guest calling sock_local_address? 

Binding with port 0 is a good call out, we probably should return the address there for consistency of the API as well.

Reply:
Ah I see. Could we perhaps cache the local address on the parent fd's `fdinfo` then? If we come via bind, then we can cache it there and avoid any extra calls in `SockAccept`. If it's a preopened socket, then we only call `SockLocalAddress` once and then return the same thing for all subsequent `SockAccept` calls?